commit fae36e81a32717ac43c0ce48702f6ff05b7cd029
Author: Michael Tremer <michael.tremer@ipfire.org>
Date:   Fri Jun 5 09:57:41 2020 +0000

    export flagged networks with their faked country names, too
    
    This will lead to some networks showing up twice. Once with
    their real country and once with their faked one.
    
    It is likely that the first one will match.
    
    Signed-off-by: Michael Tremer <michael.tremer@ipfire.org>

diff --git a/src/python/export.py b/src/python/export.py
index 69fe964..d0bbe77 100644
--- a/src/python/export.py
+++ b/src/python/export.py
@@ -23,10 +23,18 @@ import logging
 import os
 import socket
 
+import _location
+
 # Initialise logging
 log = logging.getLogger("location.export")
 log.propagate = 1
 
+flags = {
+	_location.NETWORK_FLAG_ANONYMOUS_PROXY    : "A1",
+	_location.NETWORK_FLAG_SATELLITE_PROVIDER : "A2",
+	_location.NETWORK_FLAG_ANYCAST            : "A3",
+}
+
 class OutputWriter(object):
 	suffix = "networks"
 	mode = "w"
@@ -173,6 +181,17 @@ class Exporter(object):
 				except KeyError:
 					pass
 
+				# Handle flags
+				for flag in flags:
+					if network.has_flag(flag):
+						# Fetch the "fake" country code
+						country = flags[flag]
+
+						try:
+							writers[country].write(network)
+						except KeyError:
+							pass
+
 			# Write everything to the filesystem
 			for writer in writers.values():
 				writer.finish()
